gateway/api/services/file_storage.py

```diff
@@ -0,0 +1,125 @@
+
+    Both values are being used to identify in
+        FileStorage service the path to be used
+    """
+
+    USER_STORAGE = 1
+    PROVIDER_STORAGE = 2
+
+
+SUPPORTED_FILE_EXTENSIONS = [".tar", ".h5"]
+
+logger = logging.getLogger("gateway")
+
+
+class FileStorage:  # pylint: disable=too-few-public-methods
+    """
+    The main objective of this class is to manage the access of the users to their storage.
+
+    Attributes:
+        username (str): storgae user's username
+        working_dir (WorkingDir(Enum)): working directory
+        function_title (str): title of the function in case is needed to build the path
+        provider_name (str | None): name of the provider in caseis needed to build the path
+    """
+
+    def __init__(
+        self,
+        username: str,
+        working_dir: WorkingDir,
+        function_title: str,
+        provider_name: str | None,
+    ) -> None:
+        self.file_path = None
+        self.username = username
+
+        if working_dir is WorkingDir.USER_STORAGE:
+            self.file_path = self.__get_user_path(function_title, provider_name)
+        elif working_dir == WorkingDir.PROVIDER_STORAGE:
+            self.file_path = self.__get_provider_path(function_title, provider_name)
+
+    def __get_user_path(self, function_title: str, provider_name: str | None) -> str:
+        """
+        This method returns the path where the user will store its files
+
+        Args:
+            function_title (str): in case the function is from a
+                provider it will identify the function folder
+            provider_name (str | None): in case a provider is provided it will
+                identify the folder for the specific function
+
+        Returns:
+            str: storage path.
+                - In case the function is from a provider that path would
+                    be: username/provider_name/function_title
+                - In case the function is from a user that path would
+                    be: username/
+        """
+        if provider_name is None:
+            path = self.username
+        else:
+            path = f"{self.username}/{provider_name}/{function_title}"
+
+        full_path = os.path.join(settings.MEDIA_ROOT, path)
+
+        return sanitize_file_path(full_path)
+
+    def __get_provider_path(self, function_title: str, provider_name: str) -> str:
+        """
+        This method returns the provider path where the user will store its files
+
+        Args:
+            function_title (str): in case the function is from a provider
+                it will identify the function folder
+            provider_name (str): in case a provider is provided
+                it will identify the folder for the specific function
+
+        Returns:
+            str: storage path following the format provider_name/function_title/
+        """
+        path = f"{provider_name}/{function_title}"
+        full_path = os.path.join(settings.MEDIA_ROOT, path)
```

Comment:
```suggestion
        full_path = os.path.join(settings.MEDIA_ROOT, provider_name, function_title)
```
I think this is the same.

Reply:
Maybe, let me check 👍

Reply:
Solved in [79a3f88](https://github.com/Qiskit/qiskit-serverless/pull/1529/commits/79a3f88976cfe517c17677d66b50d17fc62fa4ed)